### ISSUE TITLE
REFERENCED_BEFORE_DECLARATION error doesn't name the identifier

### ISSUE THREAD

**Issue by srawlins** [dart-lang/sdk#26902]
Analyzer currently does:

```
$ cat d.dart
void main() {
  j++;
  int j;
  print(j);
}
$ dartanalyzer d.dart
Analyzing [d.dart]...
[error] Local variables cannot be referenced before they are declared (d.dart, line 2, col 3)
1 error found.
```

As we do for many errors, I think the identifier should be named.

**Comment by srawlins** [comment]
CC @goderbauer

**Comment by srawlins** [comment]
Fixed in b5ffdbf5845784be0588457c92ec12262a7dc8dd.
